src/meta/test/AdminClientTest.cpp

```diff
@@ -0,0 +1,263 @@
+        // The third one is healthy.
+        thriftPeers.emplace_back(apache::thrift::FragileConstructor::FRAGILE, localIp, sc1->port_);
+
+        std::vector<kvstore::KV> data;
+        data.emplace_back(MetaServiceUtils::partKey(0, 1),
+                          MetaServiceUtils::partVal(thriftPeers));
+        folly::Baton<true, std::atomic> baton;
+        kv->asyncMultiPut(kDefaultSpaceId,
+                          kDefaultPartId,
+                          std::move(data),
+                          [this, &baton] (kvstore::ResultCode code) {
+            CHECK_EQ(kvstore::ResultCode::SUCCEEDED, code);
+            baton.post();
+        });
+        baton.wait();
+    }
+
+    LOG(INFO) << "Now test interfaces with retry to leader!";
+    auto client = std::make_unique<AdminClient>(kv.get());
+    {
+        LOG(INFO) << "Test transLeader, return ok if target is not leader";
+        folly::Baton<true, std::atomic> baton;
+        client->transLeader(0, 1, {localIp, sc2->port_}).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
+    {
+        LOG(INFO) << "Test member change...";
+        folly::Baton<true, std::atomic> baton;
+        client->memberChange(0, 1).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
+    {
+        LOG(INFO) << "Test add learner...";
+        folly::Baton<true, std::atomic> baton;
+        client->addLearner(0, 1).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
+    {
+        LOG(INFO) << "Test waitingForCatchUpData...";
+        folly::Baton<true, std::atomic> baton;
+        client->waitingForCatchUpData(0, 1).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
+    FLAGS_max_retry_times_admin_op = 1;
+    {
+        LOG(INFO) << "Test member change...";
+        folly::Baton<true, std::atomic> baton;
+        client->memberChange(0, 1).then([&baton](auto&& st) {
+            CHECK(!st.ok());
+            CHECK_EQ("Leader changed!", st.toString());
+            baton.post();
+        });
+        baton.wait();
+    }
+    {
+        LOG(INFO) << "Test update meta...";
+        folly::Baton<true, std::atomic> baton;
+        client->updateMeta(0, 1, HostAddr(0, 0), HostAddr(1, 1)).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+        auto peersRet = client->getPeers(0, 1);
+        CHECK(peersRet.ok());
+        auto hosts = std::move(peersRet).value();
+        ASSERT_EQ(3, hosts.size());
+        ASSERT_EQ(HostAddr(localIp, sc2->port_), hosts[0]);
+        ASSERT_EQ(HostAddr(localIp, sc1->port_), hosts[1]);
+        ASSERT_EQ(HostAddr(1, 1), hosts[2]);
+    }
```

Comment:
let's try more than 3 hosts.

Reply:
Yes. We have more situations to test. Let's open another issue to track it.